dbt/include/athena/macros/adapters/metadata.sql

```diff
@@ -65,9 +65,23 @@
 
                 )
 
-                {%- for schema in schemas -%}
-                select * from catalog where lower("table_schema") = lower('{{ schema }}')
-                {%- if not loop.last %} union all {% endif -%}
+                {%- for db, tables in schemas.items() -%}
+                  {%- for table_batch in tables|batch(100) %}
+                    select * from catalog where "table_schema" = lower('{{ db }}') 
+                    and (
+                      {%- for table in table_batch -%}
+                        "table_name" = lower('{{ table }}')
+                      {%- if not loop.last %} or {% endif -%}
+                      {%- endfor -%}
+                    )
+                    {%- if not loop.last %}
+                      union all
+                    {% endif -%}
+                  {%- endfor -%}
+
+                  {%- if not loop.last %}
+                      union all
+                  {% endif -%}
```

Comment:
Similarly just a suggestion to keep formatting the same as before to avoid less questions when this gets reviewed in dbt-athena
```suggestion
                    {%- if not loop.last %} union all {% endif -%}
                  {%- endfor -%}

                  {%- if not loop.last %} union all {% endif -%}
```

Reply:
Didn't like the way it was generating the sql. But fair changed it to inline if